### DISCUSSION TITLE
power saving mode

### DISCUSSION THREAD

**Discussion by pietrop** [vidstack/player#1013] [Q&A] [answered]
Do you currently have a way to detect if a user is on power saving mode on iOS? (and therefore autoplay would fail)

I found these
- [Detect if ios11 device is in low power mode to prevent bad UX on normally correctly autoplaying video](https://stackoverflow.com/questions/50400902/detect-if-ios11-device-is-in-low-power-mode-to-prevent-bad-ux-on-normally-correc)
- [Low power mode detection in JavaScript for iOS11?](https://stackoverflow.com/questions/46670150/low-power-mode-detection-in-javascript-for-ios11)
- https://gist.run/?id=f7b675c7a5bab86dd1488962e646c6cc

But wondering if there is a cleaner solution?

**Comment by mihar-22** [comment, marked as answer]
Hey Pietro! 

The way we handle autoplay in the library is that we call `play()` manually and look for an error to be thrown to detect failure. We don't rely on the native `autoplay` attribute, which I believe is the same for most modern player libraries. We call the `onAutoplayFail` callback in React and apply the `data-autoplay-error` attribute to the player element for styling purposes. Some more information available in our [API > Autoplay](https://www.vidstack.io/docs/player/api/autoplay?styling=default-theme) page.
